app/gui/suggestion-database/src/lib.rs

```diff
@@ -818,7 +829,7 @@ pub mod test {
     /// is [`None`].
     fn lookup_and_verify_empty_result(db: &SuggestionDatabase, fully_qualified_name: &str) {
         let lookup = db.lookup_by_qualified_name_str(fully_qualified_name);
-        assert_eq!(lookup, None);
+        assert!(lookup.is_err());
```

Comment:
should this be a downcast and `assert_matches!` instead? Just to make sure that it's the specific error kind that we expect here.